src/include/txn.i

```diff
@@ -594,11 +594,10 @@ __wt_txn_upd_visible_all(WT_SESSION_IMPL *session, WT_UPDATE *upd)
 static inline bool
 __wt_txn_upd_value_visible_all(WT_SESSION_IMPL *session, WT_UPDATE_VALUE *upd_value)
 {
-    if (upd_value->prepare_state == WT_PREPARE_LOCKED ||
-      upd_value->prepare_state == WT_PREPARE_INPROGRESS)
-        return (false);
-
-    return (__wt_txn_visible_all(session, upd_value->txnid, upd_value->durable_ts));
+    WT_ASSERT(session, upd_value->tw.prepare == 0);
```

Comment:
As we never return a prepared update, so I added an assert to ensure the same. Do we expect to return a prepared update?

Reply:
No we don't